go/vt/vtgate/engine/aggregations.go

```diff
@@ -270,17 +285,13 @@ func convertFields(fields []*querypb.Field, aggrs []*AggregateParams) []*querypb
 		if aggr.Alias != "" {
 			fields[aggr.Col].Name = aggr.Alias
 		}
-		if aggr.isDistinct() {
-			// TODO: this should move to plan time
-			aggr.KeyCol = aggr.Col
-		}
```

Comment:
as discussed, changing this on plan time is a problem. we would need to have local KeyCol index and passed in while execution.